plugins/inputs/win_services/win_services.go

```diff
@@ -117,7 +122,16 @@ func (m *WinServices) Gather(acc telegraf.Accumulator) error {
 	}
 	defer scmgr.Disconnect()
 
-	serviceNames, err := listServices(scmgr, m.ServiceNames)
+	// Create label filters if not already created
+	if !m.filtersCreated {
+		err := m.createServicesFilter()
+		if err != nil {
+			return err
+		}
+		m.filtersCreated = true
+	}
```

Comment:
let's move this to an Init() function as it only has to happen once, and then you don't need the boolean guard around it.

Reply:
Updated like suggested, Thanks!